src/Neo.VM/Types/CompoundType.cs

```diff
@@ -65,7 +66,12 @@ public sealed override bool GetBoolean()
         /// <exception cref="NotSupportedException">This method always throws the exception.</exception>
         public override int GetHashCode()
         {
-            throw new NotSupportedException();
+            var h = new HashCode();
+            h.Add(Count);
+            h.Add(Type);
+            foreach (var item in SubItems)
+                h.Add(XxHash3.HashToUInt64(item.GetSpan()));
```

Comment:
yes, I tried that. But doing _see below_ will have `ReferenceCounter` as null. Neo VM OPCodes set `ReferenceCounter`.

### All are equal, even though the last one is different.
```
itemA = new Map { [true] = false, [0] = 1 };// will be the same
itemB = new Map { [true] = false, [0] = 1 };// will be the same
itemC = new Map { [true] = false, [0] = 2 }; // will be the same
```

Reply:
> We can return always 0, or static counter, Equals will be true only when Reference is the same

Yes. There may not be a need to rewrite `GetHashCode` here, just return `Object.GetHashCode()`, because `CompundType` not overwrites `Equals`.

Reply:
> We can return always 0, or static counter, Equals will be true only when Reference is the same

Yes. There may not be a need to rewrite `GetHashCode` here, just return `Object.GetHashCode()`, because `CompundType` not overwrites Equals.

Reply:
Each item that has the same data needs the same hashcode. Object.GetHashCode() doesn't work for that.

Reply:
> Each item that has the same data needs the same hashcode. Object.GetHashCode() doesn't work for that.

`CompundType` doestnot rewrite `Equals`, so `the same data needs the same hashcode` may not be necessary.

Reply:
Hashcode is only used in the `default` un-overriden `equals` method and `hashtables`.

Reply:
> > Each item that has the same data needs the same hashcode. Object.GetHashCode() doesn't work for that.
> 
> `CompundType` doestnot override `Equals`, so `the same data needs the same hashcode` may not be necessary.

That would be true if `Object.GetHashCode` would use only the properties we want, and not all properties on the class and inherit class